templates/extras/pitfalls.md

```diff
@@ -871,3 +872,35 @@ Character | Unicode Name | How to type | Usage in Lean
 `≈` | Almost Equal To (U+2248) | `\~~`, `\approx`, `\thickapprox` | Type-dependent equivalence notation: see `HasEquiv`. Usually refers to a typeclass inferred `Setoid` instance
 `~` | Tilde (U+007E) | ASCII character | Permutation equivalence relation for `List`s, `Array`s, and `Vector`s; some special meaning for other types
 `≡` | Identical To (U+2261) | `\==` | `a ≡ b [MOD n]` means `Nat.ModEq n a b` and `a ≡ b [ZMOD n]` means `Int.ModEq n a b`. Some special meaning for other types
+
+## Default values in structure fields
+
+Users often assume that the use of `:=` in structure fields means "this field is defined
+to have this value". But this is not the case! Instead, this is the way to supply default
+values of structure fields. In particular, these values can be overridden.
+
+```
+structure foo : Type where
+  n : Nat := 37 -- I want n to always be 37...
+
+def X : foo where
+  n := 42 -- ...but that's not the way to do it
+
+#eval X.n -- 42
+
+def Y : foo where -- only if `n` is not supplied does the default value kick in.
+
+#eval Y.n -- 37
+```
+
+If you really want to have something like this, try the following instead:
+
+```
+structure foo : Type where
+
+def foo.n : Nat := 37
+
+def X : foo where
+
+#eval X.n -- 42
```

Comment:
```suggestion
#eval X.n -- 37
```

Reply:
(changing the 42s back to 37s for consistency -- "if you really want to have something like this" and then show them how to have exactly the thing they were trying to do in the failing code.